[Lin Da] iP

DALIN-Prog opened the pull request:
### Dazz
Dazz frees your mind of having to remember things you need to do. It's,

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes/))

- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download the `.jar` file from [here](https://github.com/DALIN-Prog/ip/releases).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [ ] Managing tasks (via CRUD functionality)
- [ ] Searching tasks 

##

If you Java programmer, you can use it to practice Java too. Here's the main method:
```Java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

DALIN-Prog committed:
Level-1 commit

DALIN-Prog committed:
Level-2 commit

DALIN-Prog committed:
Level-3

DALIN-Prog committed:
Level-4

DALIN-Prog committed:
A-TextUiTesting

DALIN-Prog committed:
Level-5

DALIN-Prog committed:
Level-6 and modify test-ui-test

DALIN-Prog committed:
Dazz.java: include another exception

DALIN-Prog committed:
refactor code and include enum

DALIN-Prog committed:
remove redundant exception class and refactor Dazz.java

DALIN-Prog committed:
remove redundant exception class and refactor Dazz.java

DALIN-Prog committed:
Level-7: include getter and implement save functionality

DALIN-Prog committed:
forget to include modification to Deadline,Event,Logic,Task,Todo files

DALIN-Prog committed:
Level-8: Minimal implementation

DALIN-Prog committed:
Level-8: Stretch goal

DALIN-Prog committed:
Level-7: merge Level-7 branch to master

DALIN-Prog committed:
Merge branch 'branch-Level-8'

DALIN-Prog committed:
Level-8: debug some bugs after merge branch-Level-8

DALIN-Prog committed:
make compatible for A-MoreOOP

DALIN-Prog committed:
A-MoreOOP: refactor and debug code

DALIN-Prog committed:
A-Packages: organize into packages

DALIN-Prog committed:
A-JUnit: Add JUnit Tests + make some access modifier public

DALIN-Prog committed:
.gitignore: exclude META-INF file

DALIN-Prog committed:
A-JavaDoc: Minimal

DALIN-Prog committed:
A-CodingStandard

DALIN-Prog committed:
Level-9

DALIN-Prog committed:
Merge branch 'branch-A-JavaDoc'

DALIN-Prog committed:
resolve conflict when merge A-CodingStandard

DALIN-Prog committed:
merge Level-9, resolve conflict

flairekq commented on src/main/java/dazz/Dazz.java:
```diff
@@ -0,0 +1,50 @@
+    /**
+     * Runs the chatbot, allowing it to receive user inputs
+     */
+    public void run() {
```
I like the clean code here such that at one glance I can tell what it should do

flairekq commented on src/main/java/dazz/Parser.java:
```diff
@@ -0,0 +1,97 @@
+                throw new IncompleteCommandException();
+            }
+            return new UnmarkCommand(Integer.parseInt(arr[1]));
+        case "delete":
```
Perhaps combining a few similar cases like mark, unmark and delete can help make the code shorter

flairekq commented on src/main/java/dazz/Ui.java:
```diff
@@ -0,0 +1,161 @@
+
+    /**
+     * Prints the task that has been added and the total number of tasks in the list.
+     * @param task Task task (i.e Todo, Deadline or Event)
```
Perhaps adding an empty line between the method description and parameter section improves readability

flairekq commented on src/main/java/dazz/command/DeadlineCommand.java:
```diff
@@ -0,0 +1,24 @@
+
+import java.time.LocalDateTime;
+
+public class DeadlineCommand extends Command{
```
Perhaps adding a space before { improves the readability

flairekq commented on src/main/java/dazz/exception/EmptyDescriptionException.java:
```diff
@@ -0,0 +1,5 @@
+package dazz.exception;
+
+public class EmptyDescriptionException extends DazzException {
```
I like how you further abstract the exceptions

flairekq reviewed (commented):
LGTM! Just a few nits to fix.

DALIN-Prog committed:
Merge branch 'add-gradle-support'

DALIN-Prog committed:
A-Gradle: build, run and unit tests

DALIN-Prog committed:
follow checkstyle

DALIN-Prog committed:
add GUI capability

DALIN-Prog committed:
Merge branch 'branch-Level-10'

DALIN-Prog committed:
javadocs and code refactoring for Parser.java

DALIN-Prog committed:
Practice full commit message

This commit reflects the changes to build.gradle, Ui.java and tasks.txt

build.gradle has its shadowJar component edited for testing out creating
.jar files.

Ui.java has some of it's checkstyle fixed.

tasks.txt reflects the newly added task via the GUI.

DALIN-Prog committed:
Add assertions

Modify build.gradle to allow assertions.

Assertion is added to storage as we need to ensure that the
txt file is created in order to store tasks in it.

Commands like mark, unmark, todo, delete and find are based
on single description (i.e integer/string), so we need to
make sure that it is non-empty description is passed.

DALIN-Prog committed:
Merge pull request #1 from DALIN-Prog/branch-A-Assertions

Add assertions

DALIN-Prog committed:
Follow a better code quality

Removed unused methods as they can be confusing.

Kept methods DRY.

DALIN-Prog committed:
Convert for loops to streams

Currently for-loops are used for listing tasks from tasklist.

This data flow process can be better represented using java streams
and improve code readability.

DALIN-Prog committed:
Remove unused package in Ui.java

DALIN-Prog committed:
Merge pull request #2 from DALIN-Prog/branch-A-CodeQuality

Follow a better code quality

DALIN-Prog committed:
Merge pull request #3 from DALIN-Prog/branch-A-Stream

Branch-A-stream

DALIN-Prog committed:
Fix checkstyles

DALIN-Prog committed:
Create gradle.yml

DALIN-Prog committed:
Update gradle.yml

remove I/O redirection test

DALIN-Prog committed:
Merge pull request #4 from DALIN-Prog/branch-A-CI

Create gradle.yml

DALIN-Prog committed:
Implement C-FriendlierSyntax

DALIN-Prog committed:
Implement C-Help

DALIN-Prog committed:
Merge pull request #5 from DALIN-Prog/branch-C-Friendlier-Syntax

Implement C-FriendlierSyntax

DALIN-Prog committed:
Resolve merge conflicts

DALIN-Prog committed:
Merge pull request #6 from DALIN-Prog/branch-C-Help

branch-C-Help

DALIN-Prog committed:
Resolve checkstyle issues

DALIN-Prog committed:
Implement a better GUI

Edit the format of instruction in Instruction.java

Currently there is no greeting. Also the bot and user chat label shares
the same background color. To avoid confusion, greetings from chatbot
and background color for chat label are included

DALIN-Prog committed:
Fix bug on FriendlierSyntax and resolve build conflict

DALIN-Prog committed:
Edit README.md and upload Ui.png

DALIN-Prog committed:
Update README.md

DALIN-Prog committed:
Set theme jekyll-theme-cayman

DALIN-Prog committed:
Fix minor bug

DALIN-Prog committed:
Update README.md

DALIN-Prog committed:
Change javaFxVersion to 11.0.2

DALIN-Prog committed:
Merge branch 'master' of https://github.com/DALIN-Prog/ip

DALIN-Prog committed:
Update README.md
